src/plugins/remoteClock/RemoteClockSpec.js

```diff
@@ -0,0 +1,147 @@
+
+            let clocks = openmct.time.getAllClocks();
+            remoteClock = clocks.filter(clock => clock.key === REMOTE_CLOCK_KEY)[0];
+
+            spyOn(remoteClock, '_timeSystemChange').and.callThrough();
+            spyOn(remoteClock, '_requestLatest').and.callThrough();
+            spyOn(remoteClock, '_processDatum').and.callThrough();
+            spyOn(remoteClock, 'tick').and.callThrough();
+            spyOn(openmct.telemetry, 'getMetadata').and.returnValue(metadata);
+            spyOn(openmct.telemetry, 'getValueFormatter').and.returnValue(formatter);
+            spyOn(openmct.telemetry, 'subscribe').and.callThrough();
+            spyOn(openmct.time, 'on').and.callThrough();
+            spyOn(openmct.time, 'timeSystem').and.returnValue(timeSystem);
+            spyOn(metadata, 'value').and.callThrough();
+
+            let requestPromiseResolve;
+            let requestPromise = new Promise((resolve) => {
+                requestPromiseResolve = resolve;
+            });
+            spyOn(openmct.telemetry, 'request').and.callFake(() => {
+                requestPromiseResolve([reqDatum]);
+
+                return requestPromise;
+            });
+
+            let objectPromiseResolve;
+            let objectPromise = new Promise((resolve) => {
+                objectPromiseResolve = resolve;
+            });
+            spyOn(openmct.objects, 'get').and.callFake(() => {
+                objectPromiseResolve(object);
+
+                return objectPromise;
+            });
+
+            openmct.time.clock(REMOTE_CLOCK_KEY, {
+                start: -10,
+                end: 1
+            });
+
+            Promise.all([objectPromiseResolve, requestPromise])
+                .then(done)
+                .catch(done);
+        });
+
+        it('is an available and sets up initial values and listeners', () => {
+            expect(remoteClock.key).toEqual(REMOTE_CLOCK_KEY);
+            expect(remoteClock.identifier).toEqual(TIME_TELEMETRY_ID);
+            expect(openmct.time.on).toHaveBeenCalledWith('timeSystem', remoteClock._timeSystemChange);
+            expect(remoteClock._timeSystemChange).toHaveBeenCalled();
+        });
+
+        it('will request/store the object based on the identifier passed in', () => {
+            expect(remoteClock.timeTelemetryObject).toEqual(object);
+        });
+
+        it('will request metadata and set up formatters', () => {
+            expect(remoteClock.metadata).toEqual(metadata);
+            expect(metadata.value).toHaveBeenCalled();
+            expect(openmct.telemetry.getValueFormatter).toHaveBeenCalledWith(metadataValue);
+        });
+
+        it('will request the latest datum for the object it received and process the datum returned', () => {
+            expect(openmct.telemetry.request).toHaveBeenCalledWith(remoteClock.timeTelemetryObject, REQ_OPTIONS);
+            expect(remoteClock._processDatum).toHaveBeenCalledWith(reqDatum);
```

Comment:
I tend to try and avoid creating dependencies on private functions from test specs, and instead just test external behavior. That way if the details of the class change but its behavior doesn't, the spec does not get invalidated.

I would listen to `openmct.time.on('bounds', ...)` and make sure it is called with the expected bounds (and `isTick = true`)

Reply:
Gotcha, makes sense, I'll update that!